src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -91,14 +92,16 @@ public CourseService(CourseRepository courseRepository, ExerciseService exercise
     }
 
     /**
-     * Get one course with exercises and lectures (filtered for given user)
+     * Get one course with exercises, lectures, lecture units and exams (filtered for given user)
      *
      * @param courseId the course to fetch
      * @param user     the user entity
-     * @return the course including exercises and lectures for the user
+     * @return the course including exercises, lectures, lecture units and exams for the user
      */
-    public Course findOneWithExercisesAndLecturesForUser(Long courseId, User user) {
-        Course course = courseRepository.findByIdWithLecturesAndExamsElseThrow(courseId);
+    public Course findOneWithExercisesAndLecturesAndLectureUnitsAndExamsForUser(Long courseId, User user) {
+        Course course = courseRepository.findByIdWithLecturesAndLectureUnitsAndExamsElseThrow(courseId);
+
+        filterLectureUnits(course);
         if (!authCheckService.isAtLeastStudentInCourse(course, user)) {
             throw new AccessForbiddenException("You are not allowed to access this resource");
```

Comment:
I know it's not your code here but shouldn't that be in the resource instead of the service?